libparsec/crates/client/src/certificates_ops/validate_message.rs

```diff
@@ -194,16 +194,16 @@ async fn validate_message_internal(
                 Ok(certif) => {
                     if let AnyArcCertificate::RealmRole(certif) = certif {
                         let mut as_expected = true;
-                        as_expected |= matches!(&certif.author, CertificateSignerOwned::User(certif_author) if certif_author == author);
-                        as_expected |= certif.timestamp == *timestamp;
-                        as_expected |= certif.realm_id == *id;
-                        as_expected |= certif.user_id == *ops.device.user_id();
+                        as_expected &= matches!(&certif.author, CertificateSignerOwned::User(certif_author) if certif_author == author);
+                        as_expected &= certif.timestamp == *timestamp;
+                        as_expected &= certif.realm_id == *id;
+                        as_expected &= certif.user_id == *ops.device.user_id();
```

Comment:
```suggestion
                        let mut as_expected = matches!(&certif.author, CertificateSignerOwned::User(certif_author) if certif_author == author)
                            && certif.timestamp == *timestamp
                            && certif.realm_id == *id
                            && certif.user_id == *ops.device.user_id();
```

I would prefer to have a method or a function to compare a certificate against a set of values instead.